service/matching/ack_manager_test.go

```diff
@@ -28,35 +28,50 @@ import (
 	"math/rand"
 	"testing"
 
+	"github.com/golang/mock/gomock"
+
 	"github.com/stretchr/testify/require"
-	"go.temporal.io/server/common/log"
 )
 
 func TestAckManager_AddingTasksIncreasesBacklogCounter(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+	backlogMgr := newBacklogMgr(controller)
+
 	t.Parallel()
-	mgr := newAckManager(log.NewTestLogger())
-	mgr.addTask(1)
-	require.Equal(t, mgr.getBacklogCountHint(), int64(1))
-	mgr.addTask(12)
-	require.Equal(t, mgr.getBacklogCountHint(), int64(2))
+	backlogMgr.taskAckManager.addTask(1)
+	require.Equal(t, backlogMgr.taskAckManager.getBacklogCountHint(), int64(1))
+	backlogMgr.taskAckManager.addTask(12)
+	require.Equal(t, backlogMgr.taskAckManager.getBacklogCountHint(), int64(2))
 }
 
 func TestAckManager_CompleteTaskMovesAckLevelUpToGap(t *testing.T) {
 	t.Parallel()
-	mgr := newAckManager(log.NewTestLogger())
-	mgr.addTask(1)
-	require.Equal(t, int64(-1), mgr.getAckLevel(), "should only move ack level on completion")
-	require.Equal(t, int64(1), mgr.completeTask(1), "should move ack level on completion")
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+	backlogMgr := newBacklogMgr(controller)
+
+	backlogMgr.taskAckManager.addTask(1)
 
-	mgr.addTask(2)
-	mgr.addTask(3)
-	mgr.addTask(12)
+	// Incrementing the backlog as otherwise we would get an error that it is under-counting;
```

Comment:
Sure, changes have been made. I still think it's good practice to increment the counter manually, as in this test case.

Reply:
In this case I would just change the comment to remove the part saying "otherwise we would get an error". Something like "Increment the backlog so that we don't under-count"